casper/src/main/scala/coop/rchain/casper/blocks/BlockReceiver.scala

```diff
@@ -217,7 +217,7 @@ object BlockReceiver {
 
     // Check if block should be stored
     def checkIfKnown(b: BlockMessage): F[Boolean] = {
-      val isStored = BlockStore[F].contains(b.blockHash)
+      val isStored = BlockDagStorage[F].getRepresentation.map(_.contains(b.blockHash))
```

Comment:
I'm not sure this is the correct way since you don't want to store block again in BlockStore if it is already there.